hal/watchdog_api.h

```diff
@@ -0,0 +1,156 @@
+/** \addtogroup hal */
```

Comment:
We should guard it using device_has macro like `DEVICE_WATCHDOG`

Reply:
Line 22 has it